src/improvements/SuperchainAddressRegistry.sol

```diff
@@ -137,8 +147,19 @@ contract SuperchainAddressRegistry is StdChains {
     }
 
     function saveAddress(string memory identifier, ChainInfo memory chain, address addr) internal {
-        require(addr != address(0), "Address cannot be zero");
-        require(registry[identifier][chain.chainId] == address(0), "Address already registered");
+        require(addr != address(0), string.concat("SuperchainAddressRegistry: zero address for ", identifier));
+        require(bytes(identifier).length > 0, "SuperchainAddressRegistry: empty key");
+        require(
+            registry[identifier][chain.chainId] == address(0),
+            string.concat(
+                "SuperchainAddressRegistry: duplicate key ", identifier, "for chain ", vm.toString(chain.chainId)
+            )
+        );
+        // TODO We should be able to have this check, but currently tests revert if uncommented.
+        // require(
+        //     bytes(addressInfo[addr].identifier).length == 0,
+        //     string.concat("SuperchainAddressRegistry: address already registered for ", vm.toString(addr))
+        // );
```

Comment:
It seems the existing addresses like `FoundationUpgrades` safe get added for each chain, but instead they should just be added once under the sentinel key. I think the root `addresses.toml` should get read in during construction of both address registries and saved off. For this contract we'd save it with the sentinel key, and need to make sure to save off the right section based off `block.chainid`. But this can come in a future PR